src/server/router/index.js

```diff
@@ -26,11 +26,17 @@ const mapRouteToPageTitle = (path) => {
 };
 
 const renderMarkup = (store, routerProps) => {
-  const application = renderToString(
-    <Provider store={store}>
-      <RouterContext {...routerProps}/>
-    </Provider>
-  );
+  let application;
+
+  try {
+    application = renderToString(
+      <Provider store={store}>
+        <RouterContext {...routerProps}/>
+      </Provider>
+    );
+  } catch (error) {
+    console.log(error);
```

Comment:
We should throw a proper 500 error at this state. This is a gradual improvement over the original code when nothing was happening.